test/cmdline_test.rb

```diff
@@ -21,4 +21,60 @@ def initialize
       expect(subject.ShowHandler("id" => "0")).to eq true
     end
   end
+
+  describe "AddHandler" do
+    let(:options) { { "name" => "vlan0", "ethdevice" => "eth0", "bootproto" => "dhcp" } }
+
+    before do
+      allow(Yast::Report).to receive(:Error)
+      allow(Yast::LanItems).to receive(:Commit)
+    end
+
+    context "when called without type" do
+      context "and it cannot be infered from the given options" do
+        it "reports an error" do
+          expect(Yast::Report).to receive(:Error)
+          subject.AddHandler(options.reject { |k| k == "ethdevice" })
```

Comment:
np: In order to reduce duplicity, you could move the `#reject` logic to a *let* block.